worker/src/RTC/Router.cpp

```diff
@@ -2,6 +2,7 @@
 // #define MS_LOG_DEV_LEVEL 3
 
 #include "RTC/Router.hpp"
+#include "DepLibUring.hpp"
```

Comment:
I'm a bit worried about having to wrap every call to `DepLibUring` into a `#ifdef MS_LIBURING_SUPPORTED` block. IMHO `DepLibUring.hpp` should ALWAYS expose the same class and should only include `liburing.h` if the define is given. And it should expose a `IsSupported()` static method that returns true/false depending on that macro.

Reply:
So you are asking to change:

```
#ifdef #MS_LIBURING_SUPPORTED
```

by:

```
if (DepLibUring::IsSupported())
````

is that it?

Reply:
Yes. So only `DepLibUring::IsSupported()` method checks the `MS_LIBURING_SUPPORTED` given macro. Wdyt? Personally I don't like having all those `#if macro ... #end`. It's not readable. Or it's not real code.

Reply:
The thing with the macro is that MacOS, Windows and binaries built with a linux kernel version <6 will not even have that symbol in the binary, and hence there is no code to run. Such code will not exist, simply. lf we put the macro behind a function everyone will call it even if liburing is not even present in the binary.

We are not going to burn the CPU for calling an empty method **every time** we send **anything** to the network.., but we can avoid making the call when the feature does not even exist in the binary by using a macro.

There are 3 places in the code making use of the macro: Router.cpp, UdpSocketHandler.cpp and TcpConnectionHanlder.cpp. Maybe we'll end up using liburing for every single thing we send over the network which will remove the usage in Router.cpp. It's two places where this macro is used.

Reply:
> Why don't you like this?

I don't find it bad using a macro not to get code compiled in architectures that is not going to be used. And as said in the previous comment, I don't find a reason to call a method in architectures and older linux machines that we know it will have no effect.

Any library out there has macros when it deals with multiple architectures and we are a library. This feature only works for linux.

Having said this, if given my rationale you insist on putting the macro inside a function that everyone will call then I will do it.

Reply:
> I don't find a reason to call a method in architectures and older linux machines that we know it will have no effect.

With my proposal the only method that would be called in NON supported platforms will be `DepLibUring::IsSupported()`.

> Having said this, if given my rationale you insist on putting the macro inside a function that everyone will call then I will do it.

No, I don't want to insist or mandate this change. I'm just exposing my concern. Let me check again how the code looks as it is.